Sources/StreamChatUI/ChatMessageList/ChatMessageListCollectionView.swift

```diff
@@ -5,13 +5,112 @@
 import StreamChat
 import UIKit
 
+/// Protocol that adds delegate methods specific for `ChatMessageListCollectionView`
+public protocol ChatMessageListCollectionViewDelegate: UICollectionViewDelegate {
+    /// Get date for item at given indexPath
+    /// - Parameters:
+    ///   - collectionView: CollectionView requesting date
+    ///   - indexPath: IndexPath that should be used to get date
+    func collectionView(_ collectionView: ChatMessageListCollectionView, dateForItemAt indexPath: IndexPath) -> Date?
+}
+
 /// The collection view that provides convenient API for dequeuing `_СhatMessageCollectionViewCell` instances
 /// with the provided content view type and layout options.
-open class ChatMessageListCollectionView: UICollectionView {
+open class ChatMessageListCollectionView: UICollectionView, Customizable, ComponentsProvider {
+    // Class does not use any ExtraData generic so we just need to make compiler happy
+    public typealias ExtraData = NoExtraData
```

Comment:
Even though the compiler is happy our customers with custom extra data types won't be 😅 The `ExtraData` here must match other extra data of the client. So this should rather be:
```
open class ChatMessageListCollectionView<ExtraData: ExtraDataTypes>: UICollectionView, ... {}
```

Reply:
So if I want to inject non-generic view from `ComponentsProvider` to another non-generic view, it is necessary to make both (and possibly all custom superviews) generic?